constraintsolver/src/CGSolver.cpp

```diff
@@ -152,17 +152,17 @@ bool CGSolver::getSolution(vector<Variable *> &vars, vector<shared_ptr<ProblemDe
 
     for (auto &c : calls)
     {
-        auto constraintTerm = dynamic_pointer_cast<autodiff::Term>(c->getConstraint());
-        if (!constraintTerm)
+        std::shared_ptr<autodiff::Term> constraintTerm = dynamic_pointer_cast<autodiff::Term>(c->getConstraint());
+        if (constraintTerm.get() == nullptr)
```

Comment:
Isn't that a problem with regard to the "Note" on this page here: http://en.cppreference.com/w/cpp/memory/shared_ptr/operator_bool

Reply:
iIt appears that way. In that case, the operator bool() has the same problem though.
In practice this works.